src/components/Scope/index.js

```diff
@@ -41,11 +42,18 @@ export const Scope = ({ children, filterPlaceholder }) => {
 	const name = "scopeSelector";
 	const [currentScope, defaultNodeState, getScope] = useScopeData();
 	const [{ show = false, nodeState, filter }, updateViewState] = useViewState(name);
+	usePreviousModified(
+		show,
+		current =>
+			current && updateViewState("nodeState", { ...nodeState, ...defaultNodeState }),
```

Comment:
You end up missing some dependencies for useEffect here. If defaultNodeState changes, the effect function here won't be updated - `usePreviousModified` will keep using the one it was passed with at first. This means there's potential for ugly bugs down the line.

I'd advise a useEffect hook here which depends on both show, nodeState and defaultNodeState, and resets the nodestate if show is false. Something like:
```js
useEffect(() => {
  if (show) updateViewState(<as above>);
}, [show, nodeState, defaultNodeState]);
```

Reply:
Well, after some investigation, it is a "no can't do" unfortunately. Calling "updateViewState" within the useEffect would update the state over and over (as you know already I'm sure) and using the nodeState object in the array of the second parameter not prevent the useEffect to be triggered as it will be different that the previous time everytime we call "updateViewState". It does not perform a deep object comparison. Here are two urls about that issue and some suggestions are made but I am not sure we want to go that path as for now. I am trying to find another solution but I would like your feedback first.

https://stackoverflow.com/questions/55808749/use-object-in-useeffect-2nd-param-without-having-to-stringify-it-to-json/55808960?noredirect=1#comment98705815_55808960

https://stackoverflow.com/questions/53601931/custom-useeffect-second-argument

Reply:
I think it could be made to work. I'll dig into it further in the morning exactly how. I do think your hook may work for it, but it needs the right effect function, potentially a useCallback-wrapped one. (But it's evening here now, so I'm not at my very sharpest. :relaxed:)

Reply:
Oh, I have no doubt there is a solution and I also thought about a callback method. Let's talk about it tomorrow morning. I was not expecting an answer from you so soon btw. Have a good evening.

Reply:
Okay, here's what I came up with: Set the default node state on the TreeView component as it was before. When `viewState.show` turns true, set `viewState.nodeState`  to {}. This deletes the previous node state, and the default node state takes over.

I'll add some comments in the code diff to clear this up.

Reply:
There. This also means you don't need the current `viewstate.nodeState`, since all you do it reset it.

Reply:
Hi Gert, I understand perfectly what you are saying, however, like I told you last week, I feel kind of sad that we lose the feature to have expanded branches somewhere else in the scope tree if I set viewState.nodeState to {}... just saying. Allow me to do one more suggestion before we come to that please. See my next push before to comment (unit tests will be fixed later, after comments and/or modifications). Basically, I only call the action in my hook when the predicate returns true, that is, when the selector is opening.

Now, IMHO, if a set nodeState to {} everytime we open the selector, I don't see the purposes anymore to keep that state in the REDUX store. I have the feeling that useState() would be enough for the Scope component. Again, don't get me wrong please, I am not trying to argue here, I am only trying to state how I understand things, whether I am right or wrong. If the latter, I would like to know your reason to understand better.

Reply:
I'll resolve this conversation, awaiting your update.